### PR TITLE
Implement Server Side Rendering

### PR DESCRIPTION
**Pull request by christulin**

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
+import express from'express';
 const app = express();
-const axios = require('axios');
-const fallback = require('./fallbacks.js');
-const routes = require('./routes');
+import axios from 'axios';
+import fallback from './fallbacks.js';
+import routes from './routes.js';
+import React from 'react';
+import ReactDOMServer from 'react-dom/server.js';
```
Comment (amosNash) [inline server/app.js:9]: Why make the change from require to import?

File: server/app.js [hunk 2]
```diff
@@ -1,13 +1,35 @@
...
+// app.use('/rooms/:id', express.static('public'));
+
+app.get('/rooms/:id', (req, res) => {
+  const app = ReactDOMServer.renderToString(<App />);
+
+  const indexFile = path.resolve('./build/index.html');
+  fs.readFile(indexFile, 'utf8', (err, data) => {
+    if (err) {
+      console.error('Something went wrong:', err);
+      return res.status(500).send('Oops, better luck next time!');
```
Comment (amosNash) [inline server/app.js]: Nice error message haha

File: server/app.js [hunk 3]
```diff
@@ -1,13 +1,35 @@
...
+import React from 'react';
+import ReactDOMServer from 'react-dom/server.js';
+// import App from '../client/src/components/App.js';
+
 
 app.use('/', express.static('public'));
-app.use('/rooms/:id', express.static('public'));
+// app.use('/rooms/:id', express.static('public'));
+
+app.get('/rooms/:id', (req, res) => {
```
Comment (amosNash) [inline server/app.js]: This looks good, I like how clean the code is!

File: server/server.js [hunk 4]
```diff
@@ -2,8 +2,18 @@ import app from './app.js';
 const PORT = process.env.PORT || 5000;
 import ssr from './ssr.js';
 import newrelic from 'newrelic';
+const cache = require('express-redis-cache')();
```
Comment (amosNash) [inline server/server.js:5]: I would look into express-redis-cache. I implemented caching another way but this looks to be very simple.

File: server/server.js [hunk 5]
```diff
@@ -2,8 +2,18 @@ import app from './app.js';
...
-app.get('/rooms/:id', ssr);
+app.get('/rooms/:id',
+  (req, res, next) => {
+    const id = req.params.id;
+    if (id) {
+      res.express_redis_cache_name = id;
+    }
+    next();
+  },
+  cache.route({ expire: 50000 }),
```
Comment (amosNash) [inline server/server.js:15]: Nice to add LRU cache strategy.

**Review by amosNash — COMMENTED** [review]
Code looks good!

File: server/app.js (hunk 1 above)
Comment (christulin) [reply server/app.js:9]: it was required with way I set up SSR

**Pull request merged**